cmd/kubemark/hollow-node.go

```diff
@@ -181,52 +185,84 @@ func run(config *hollowNodeConfig) {
 	if len(config.TenantServers) == 0 {
 		klog.V(3).Infof("TenantServers is not set. Default to single tenant partition and clientConfig setting")
 		config.TenantServers = make([]string, 1)
-		config.TenantServers[0] = clientConfigs.GetConfig().Host
+		config.TenantServers[0] = config.KubeconfigPath
 	}
 	if config.ResourceServer == "" {
 		klog.V(3).Infof("Resource is not set. Default to clientConfig setting")
-		config.ResourceServer = clientConfigs.GetConfig().Host
+		config.ResourceServer = config.KubeconfigPath
 	}
 
 	// initialize the kubeclient manager
 	kubeclientmanager.NewKubeClientManager()
 
+	// TODO: set this from the hollow node env variable
+	enableInSecurePorts := false
+
 	numberTenantPartitions := len(config.TenantServers)
-	clients := make([]clientset.Interface, numberTenantPartitions)
-	for i := 0; i < len(clients); i++ {
-		tenantCfg := restclient.CopyConfigs(clientConfigs)
-		for _, cfg := range tenantCfg.GetAllConfigs() {
-			cfg.Host = config.TenantServers[i]
-		}
-		clientFromConfig, err := clientset.NewForConfig(tenantCfg)
-		if err != nil {
-			klog.Fatalf("Failed to create a ClientSet: %v. Exiting.", err)
+	client := make([]clientset.Interface, numberTenantPartitions)
+	for i := 0; i < len(client); i++ {
+		if enableInSecurePorts == false {
```

Comment:
So in secure mode, we use kubeconfig; in non secure mode, we use host address directly?

Reply:
Current decision is no. once we moved to kubeconfig as the input to the components, the insecure host will be simply the "server: http://servername:port" in the config.

Reply:
I would suggest keep it simple. Just let kubeconfig to handle both secure and insecure mode. No need to support multiple args for this case

Reply:
i agree. if we all think we move forward to use kubeconfig. this will be dead code once we moved on. we can either clean up now or later on.

Reply:
the flag is removed after the refactor to remove old URL based clientset creation